src/deploy/functions/validate.ts

```diff
@@ -85,4 +86,32 @@ export function checkForInvalidChangeOfTrigger(
       )}] Scheduled functions cannot be changed to event handler or HTTP functions`
     );
   }
+
+  if (changedV2PubSubTopic(exFn, fn)) {
+    throw new FirebaseError(
+      `[${getFunctionLabel(fn)}] Cannot change the Pub/Sub topic of a v2 Cloud Function`
+    );
+  }
+}
+
+function changedV2PubSubTopic(exFn: backend.FunctionSpec, fn: backend.FunctionSpec): boolean {
```

Comment:
Should we delete and recreate with the new topic in this case? Or should we require a remove, deploy, re-add, deploy as this would (I think)?

Reply:
If someone is using `firebase deploy` in a CI/CD environment, this means that if they want to redeploy the exact same function they had before but with a new topic, they would need to comment out, commit/push, uncomment, commit/push.

Even though the backend doesn't allow changing topic, we could detect a topic change as similar to redeploying a different trigger to the same function name and just delete the old one and create the new one. That seems more consistent with our behavior elsewhere.

Reply:
Are you sure it's wise to risk deleting the function and then failing to recreate it? It's a bit of a PITA to have some functions that need to be deleted before created and other functions that just get deleted outright after creates are done, but I probably need that to upgrade a scheduled function from Pub/Sub to HTTPS anyway

Reply:
On one hand I'm not 100% sure this is a great idea, but OTOH it lays the foundation for upgrading scheduled functions, so I'm onboard I guess. Additional changes incoming